Modules/xASL_module_ASL.m

```diff
@@ -58,7 +58,14 @@
 %% Admin
 result = false;
 [x] = xASL_init_SubStructs(x);
-x = xASL_init_InitializeMutex(x, 'ASL'); % starts mutex locking process to ensure that everything will run only once
+[x, bLocked] = xASL_init_InitializeMutex(x, 'ASL'); % starts mutex locking process to ensure that everything will run only once
```

Comment:
I understand what are you trying to do and I agree that we should make sure that we don't execute processing even if another subpart of the subject is locked. 

BUT, you are not doing that correctly. You have your MUTEX function that should take care of all that. 

And that mutex should either create a nice lock and let you through. Or it should report an error and not let you through. This was nicely programmed by Paul. You have tried to hack it and it would work, but you are sort of delegating MUTEX functionality outside of the mutex. Which goes against how mutex should be used. The mutex should really control everything. But you are now taking its functionality outside and letting other functions to decide if they will respect that or not. 

So I agree with what you are doing in general, but the code has to change. This function should stay as it was, no bLocked. This should all be handled by the mutex itself. In the same way as if a specific module was locked - it will simply refuse to lock again and skip it.

Reply:
Are you sure? `xASL_GoNoGo` (== what you call the MUTEX function) is module- and subject-ignorant. It only handles the current module and folder.

So now `xASL_GoNoGo` is only the function for single module mutex handling.
`xASL_init_InitializeMutex` is the wrapper around this, dealing with subject and other modules.

Your suggestion to put both inside `xASL_GoNoGo` would combine the wrapper and function, to me this seems as more dirty programming (less layered/modular). Perhaps, there is something I miss that could be saving a bit code and debugging with the object-oriented programming of `xASL_GoNoGo`, which could be the benefit of expanding `xASL_GoNoGo`, is that what you mean?

But I would not mix up the "mutex function" `xASL_GoNoGo` and "mutex subject/module wrapper" `xASL_init_InitializeMutex`.

Reply:
Sorry. I might have missed on a few details. But I hope I got the overall message right. I didn't want to go into details of what is happening inside `xASL_GoNoGo` or inside `xASL_init_initializeMutex`. But my point was that some MUTEX function should either lock the folder or say it can't be locked, but all that locking should happen modularily inside it and it shouldn't output any variables outside to be kept and used by the user. So the whole process should be screened from the user. 

I do not know how this exactly works inside these functions, but seeing that you have `bLocked` that you have to deal with yourself outside of the MUTEX wasn't clean. So adding this to `x.mutex` sounds like a solution that will make me happy :)

I guess we are now on a similar note. Try to adapt the code and I'll then go step by step through it and verify it.

Reply:
Now throughout `bLocked` -> `x.mutex.bAnyModuleLocked`

Reply:
Sounds good. Will check and test in detail.